### PR TITLE
Improve sync storage, add storage diagnostics to settings

### PR DESCRIPTION
**Pull request by OsaSoft**
## Summary

This PR fixes a critical bug where videos marked as watched on one device weren't syncing to other devices, even though the "Test Sync" diagnostic passed.

### Root Cause

1. **No error detection on `storage.sync.set()`** - When sync quota was exceeded, `brwsr.runtime.lastError` was set but never checked. The code assumed success even when data wasn't saved.

2. **Stale batch keys not cleaned up** - When fewer batches were needed after clearing videos, old `vw_*` keys remained in sync storage and got reloaded.

3. **Inverted timestamp ordering on load** - When loading videos from sync on another device, timestamps were assigned in reverse order, causing newer videos to appear older.

4. **No way to free up quota** - Users had no mechanism to clear old videos when sync storage was full (100KB limit).

#### Bug Fixes (common.js)
- Added `brwsr.runtime.lastError` checking in `syncWatchedVideos()` to detect sync failures
- Added cleanup of stale batch keys after syncing (removes unused `vw_*` keys)
- Fixed timestamp assignment in `loadWatchedVideos()` - newest videos now correctly get highest timestamps
- Added `clearOldestVideos(count)` function to remove oldest videos and free quota
- Added `lastSyncError` tracking for diagnostics

#### Settings Page Improvements (settings.js, settings.html, settings.css)
- Added "Clear oldest videos" UI with configurable count input
- Added quota warning when local video count exceeds sync count
- Color-coded storage usage display (red >95%, orange >80%)
- Fixed `performClearVideos()` to properly clear local storage (was using wrong key pattern)
- Added chunked removal for large numbers of keys
- Added proper error handling to storage remove operations
- Fixed file input overflow issue (long filenames no longer hide Import button)

#### Tests (tests/unit/sync.test.js)
Added 22 comprehensive tests covering:
- `syncWatchedVideos` - batching, sorting, stale key cleanup, validation
- `loadWatchedVideos` - sync loading, timestamp ordering, data merging
- `clearOldestVideos` - removal, sync updates, edge cases
- `saveVideoOperation` - storage operations
- Cross-device sync scenarios - verifies ordering is preserved across multiple sync cycles
- Stale batch cleanup scenarios

## Test plan

- [x] All 104 unit tests pass
- [ ] Test on Firefox: Mark video as watched on PC1, verify it appears on PC2
- [ ] Test "Clear oldest videos" reduces sync storage usage
- [ ] Test quota warning appears when sync is near capacity
- [ ] Verify "Clear Data" properly clears both local and sync storage

### Before (Bug)
```javascript
// No error checking - silent failures
await new Promise(resolve => {
    brwsr.storage.sync.set(batches, resolve);
});
```

### After (Fixed)
```javascript
// Proper error detection
await new Promise((resolve, reject) => {
    brwsr.storage.sync.set(batches, () => {
        if (brwsr.runtime.lastError) {
            reject(new Error(brwsr.runtime.lastError.message));
        } else {
            resolve();
        }
    });
});

// Cleanup stale batch keys
const keysToRemove = existingBatchKeys.filter(key => !newBatchKeys.includes(key));
if (keysToRemove.length > 0) {
    await brwsr.storage.sync.remove(keysToRemove);
}
```

### Timestamp Fix
```javascript
// Before (WRONG - newest got lowest timestamp)
const now = Date.now() - operations.length;
for (const [index, operation] of operations.entries()) {
    saveVideoOperation(operation, now + index);
}

// After (CORRECT - newest gets highest timestamp)
const now = Date.now();
for (const [index, operation] of operations.entries()) {
    saveVideoOperation(operation, now - index);
}
```

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -161,16 +168,72 @@ async function syncWatchedVideos() {
...
+        }
+
         lastSyncUpdate = Date.now();
+        lastSyncError = null;
     }
     catch (error) {
-        logError({message: "Sync failed", stack: error?.stack || ""});
+        lastSyncError = error.message;
+        logError({message: "Sync failed: " + error.message, stack: error?.stack || ""});
     }
```
[the remark this answers is not in the document]
Comment (OsaSoft) [reply common.js]: Fixed in 3b61261. The catch block now re-throws the error after logging, so callers like `forceSyncAll` will catch it and display the appropriate error message.

File: common.js [hunk 2]
```diff
@@ -161,16 +168,72 @@ async function syncWatchedVideos() {
...
+        .sort((a, b) => watchedVideos[a] - watchedVideos[b]);  // Oldest first
+
+    const toRemove = sortedByAge.slice(0, count);
+
+    for (const key of toRemove) {
+        delete watchedVideos[key];
+    }
+
+    // Remove from local storage
+    await new Promise(resolve => brwsr.storage.local.remove(toRemove, resolve));
```
[the remark this answers is not in the document]
Comment (OsaSoft) [reply common.js]: Fixed in 3b61261. Added `lastError` check that logs the error but still resolves (non-blocking cleanup), consistent with the suggested approach.
